### PR TITLE
indexer,rest: include input as well as output objects for handlers

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-rest-api/src/checkpoints.rs [hunk 1]
```diff
@@ -77,44 +80,79 @@ pub async fn get_full_checkpoint(
...
+    pub transactions: Vec<CheckpointTransaction>,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct CheckpointTransaction {
+    pub transaction: Transaction,
+    pub effects: TransactionEffects,
+    pub events: Option<TransactionEvents>,
+    pub input_objects: Vec<Object>,
+    pub output_objects: Vec<Object>,
```
Comment (dariorussi) [inline crates/sui-rest-api/src/checkpoints.rs:155]: do I understand correctly that input objects are the objects before execution and, if modified, their after execution would be in the output objects as well?

**Review by dariorussi — COMMENTED** [review]
just a comment for now to give others an opportunity to comment and I'll come back tomorrow to approve, if not approved already

File: crates/sui-rest-api/src/checkpoints.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-rest-api/src/checkpoints.rs:155]: correct. I'll make a comment to this effect for clarity

**Review by dariorussi — APPROVED** [review]
accepting as promised ;) and also we can change things as needed.
I will update the data analytics code to use the new `CheckpointTransaction` once this goes in. 
I am not too clear about having the entire input objects there as opposed to just the object ref, but maybe there are important usages for that

**Review by longbowlu — APPROVED** [review]

**Pull request merged**